src/main/java/org/folio/services/validator/engine/ValidationEngineServiceImpl.java

```diff
@@ -162,23 +164,22 @@ private void validatePasswordByRexExpRule(final String password,
     Promise<JsonObject> promise = Promise.promise();
     String okapiUrl = headers.get(OKAPI_URL_HEADER);
     String userNameRequestUrl = String.format("%s/users?query=id==%s", okapiUrl, userId);
-    HttpClientRequest request = httpClient.getAbs(userNameRequestUrl);
+    HttpRequest<Buffer> request = webClient.getAbs(userNameRequestUrl);
     request
       .putHeader(OKAPI_HEADER_TOKEN, headers.get(OKAPI_HEADER_TOKEN))
       .putHeader(OKAPI_HEADER_TENANT, headers.get(OKAPI_HEADER_TENANT))
       .putHeader(HttpHeaders.CONTENT_TYPE.toString(), MediaType.APPLICATION_JSON)
       .putHeader(HttpHeaders.ACCEPT.toString(), MediaType.APPLICATION_JSON)
-      .handler(response -> {
-        if (response.statusCode() != 200) {
-          response.bodyHandler(buf -> {
+      .send(ar -> {
+        if (ar.succeeded()) {
```

Comment:
To reduce nesting and indentation and improve readability you may use the guard clause pattern:

    if (ar.failed()) {
      return;
    }

Reply:
Should it fail the promise?

Reply:
This if-clause doesn't have an else-clause and therefore neither fails nor completes the promise if ar has failed. Is this intended?

Reply:
Added promise failure